Firestore/core/src/firebase/firestore/util/status.h

```diff
@@ -110,20 +116,37 @@ class ABSL_MUST_USE_RESULT Status {
  private:
   static const std::string& empty_string();
   struct State {
+    State() = default;
+    State(const State& other);
+
     FirestoreErrorCode code;
     std::string msg;
+    std::unique_ptr<PlatformError> wrapped;
   };
-  // OK status has a `NULL` state_.  Otherwise, `state_` points to
+  // OK status has a `nullptr` state_.  Otherwise, `state_` points to
   // a `State` structure containing the error code and message(s)
   std::unique_ptr<State> state_;
 
   void SlowCopyFrom(const State* src);
 };
 
+class PlatformError {
```

Comment:
Very optional: this interface doesn't really have any behavior. It's just an opaque handle that can only be used after casting, plus deep copy semantics. Would `absl::any` fit the bill?

Reply:
I've added some behavior to this so `absl::any` doesn't work as well now.